src/Storages/StorageBuffer.cpp

```diff
@@ -996,8 +996,11 @@ void StorageBuffer::reschedule()
         std::unique_lock lock(buffer.tryLock());
         if (lock.owns_lock())
         {
-            min_first_write_time = buffer.first_write_time;
-            rows += buffer.data.rows();
+            if (buffer.data)
```

Comment:
@DimasKovas can you shed some light on this? What for this patch?

Reply:
Sure.

This code calculates the minimum `first_write_time` among all buffers (i.e. the timestamp of the first unflushed row in the StorageBuffer) and then uses it to schedule the next flush invocation.

But as you can see, the previous code didn't calculate the minimum - it just assigned `min_first_write_time` to  `buffer.first_write_time` on every iteration, so it just ignored all the buffers but the last. Moreover, it ignored the fact that a buffer may be empty. In that case `first_write_time` is 0 and doesn't actually make sense.

As a result, the next flush invocation in previous code could be scheduled earlier than expected (when `min_first_write_time` became 0) or later than expected (when `min_first_write_time` isn't 0, but isn't the minimum as well)

This patch tries to fix this issue and calculate `min_first_write_time` properly